crates/cli/src/commands.rs

```diff
@@ -73,9 +74,121 @@ pub struct CompileAndBuildCommandOpts {
     pub no_source_compression: bool,
 }
 
+#[derive(Debug, Parser)]
+pub struct BuildCommandOpts {
+    #[arg(value_name = "INPUT", required = true)]
+    /// Path of the JavaScript input file.
+    pub input: PathBuf,
+
+    #[arg(short, default_value = "index.wasm")]
+    /// Desired path of the WebAssembly output file.
+    pub output: PathBuf,
+
+    #[arg(
+        short = 'C',
+        long_help = "Available codegen options:
+-C dynamic[=y|n] -- Creates a smaller module that requires a dynamically linked QuickJS provider Wasm module to execute (see `emit-provider` command).
+-C wit=path -- Optional path to WIT file describing exported functions. Only supports function exports with no arguments and no return values.
+-C wit-world=val -- Optional WIT world name for WIT file. Must be specified if WIT is file path is specified.
+-C no-source-compression[=y|n] -- Disable source code compression, which reduces compile time at the expense of generating larger WebAssembly files.
+    "
+    )]
+    /// Codegen options.
+    pub codegen: Vec<CodegenOption>,
+}
+
 #[derive(Debug, Parser)]
 pub struct EmitProviderCommandOpts {
     #[structopt(short, long)]
     /// Output path for the provider binary (default is stdout).
     pub out: Option<PathBuf>,
 }
+
+#[derive(Clone, Debug, Parser)]
+pub struct CodegenOptionGroup {
+    /// Creates a smaller module that requires a dynamically linked QuickJS provider Wasm
+    /// module to execute (see `emit-provider` command).
+    pub dynamic: bool,
+    /// Optional path to WIT file describing exported functions.
+    /// Only supports function exports with no arguments and no return values.
+    pub wit: Option<PathBuf>,
+    /// Optional path to WIT file describing exported functions.
+    /// Only supports function exports with no arguments and no return values.
+    pub wit_world: Option<String>,
+    /// Disable source code compression, which reduces compile time at the expense of generating larger WebAssembly files.
+    pub no_source_compression: bool,
+}
+
+#[derive(Clone, Debug)]
+pub enum CodegenOption {
+    /// Creates a smaller module that requires a dynamically linked QuickJS provider Wasm
+    /// module to execute (see `emit-provider` command).
+    Dynamic(bool),
+    /// Optional path to WIT file describing exported functions.
+    /// Only supports function exports with no arguments and no return values.
+    Wit(PathBuf),
+    /// Optional path to WIT file describing exported functions.
+    /// Only supports function exports with no arguments and no return values.
+    WitWorld(String),
+    /// Disable source code compression, which reduces compile time at the expense of generating larger WebAssembly files.
+    NoSourceCompression(bool),
```

Comment:
I had thought of renaming this to `SourceCompression(bool)` and making it enabled by default. Mostly because I feel that it becomes easier to reason about as a boolean flag.